templates/base.html

```diff
@@ -7,43 +7,56 @@
   {% block htmlhead %}
 
   <!-- Page Info -->
-  <title>{% block title %}Home{% endblock %} | OddBird</title>
+  <title>{% block title %}Home{% endblock title %} | OddBird</title>
   <meta name="description" content="{% block description %}Custom-built web tools for humans{% endblock description %}" />
 
   <!-- Site Info -->
-  {% block robots %}<meta name="robots" content="{{ 'noindex, nofollow' if config.public == false else 'index, follow' }}" />{% endblock %}
+  {% block robots %}<meta name="robots" content="{{ 'noindex, nofollow' if config.public == false else 'index, follow' }}" />{% endblock robots %}
   <meta name="author" content="OddBird" />
   <meta name="keywords" content="custom web development, web tools, open source, python, django, sass, scss, susy, javascript, jquery, js" />
   <meta name="language" content="en-us" />
 
   {% block feedlinks %}
-  <link href="/feed.atom" rel="alternate" title="OddBird blog posts" type="application/atom+xml">
-  {% endblock %}
+  <link href="/feed.atom" rel="alternate" title="OddBird blog posts" type="application/atom+xml" />
+  {% endblock feedlinks %}
 
   {%- for link in links %}
-    <link rel="{{ link.rel }}" href="{{ link.href }}"{% if link.media %} media="{{ link.media }}"{% endif %} type="{{ link.type }}">
+    <link rel="{{ link.rel }}" href="{{ link.href }}"{% if link.media %} media="{{ link.media }}"{% endif %} type="{{ link.type }}" />
   {%- endfor %}
 
   <!-- Microformat Profiles -->
   <!-- hCard, hCalendar, rel-license, rel-tag, XOXO, XFN, VoteLinks -->
   <link rel="profile" href="http://microformats.org/profile/specs/" />
 
   <!-- Always force latest IE rendering engine (even in intranet) & Chrome Frame -->
-  <meta http-equiv="cleartype" content="on">
+  <meta http-equiv="cleartype" content="on" />
   <meta http-equiv="X-UA-Compatible" content="IE=edge;chrome=1" />
 
   <!-- http://t.co/dKP3o1e -->
-  <meta name="HandheldFriendly" content="True">
-  <meta name="MobileOptimized" content="320">
-  <meta name="viewport" content="initial-scale=1">
+  <meta name="HandheldFriendly" content="True" />
+  <meta name="MobileOptimized" content="320" />
+  <meta name="viewport" content="initial-scale=1" />
+
+  <!-- Open Graph -->
+  {% set image = config.image[0].src if config.image else 'oddsite.jpg' %}
+  {% set image = image if '://' in image else [config.canonical_url, 'static/images/blog/', image]|join('') %}
+  <meta property="og:title" content="{{ self.title() }} | OddBird" />
+  <meta property="og:type" content="website" />
+  <meta property="og:image" content="{{ image }}" />
+  <meta property="og:url" content="{% block og_url %}{{ config.canonical_url }}{% endblock og_url %}" />
```

Comment:
So L42 is kinda ugly, but it works. Here on L46, I'm defining a `og_url` block and setting `canonical_url` (which is http://oddbird.net/) as the default/fallback. Then in `templates/rst_display.html#L8` I'm overriding the block with the actual URL of the individual page. It works for blog pages, but breaks for pages like `/contact/` and `/blog/` and `/author/kit/` etc.

Ideally this could be handled by a generic global `get_current_url()` fn or something, but I don't know if that's feasible. If the eventual solution can be called the same way everywhere, then it should just happen here. If it needs to be called differently (or passed different arguments) in various places, then we need to use a `block` here which gets overridden elsewhere (as it is currently).